waku.md

```diff
@@ -0,0 +1,377 @@
+
+It is desirable to have a strategy for maintaining forward compatibility between `waku/0` and future version of waku. Here we outline some concerns and strategy for this.
+
+<!-- TODO: Outline difference between _bridging_ and data format -->
+
+<!-- TODO: Think about how to maintain forwards capability for waku/v0 -> v1 -> v2, etc. -->
+<!-- Example user story: changing version number to 1; moving to libp2p; changing routing to PSS style; remove PoW; replacing PoW with zkSNARKs; adding packet codes for rate limit / accounting for resources feedback; additional disconnect features -->
+
+<!-- TODO: Right now we have
+
+    if m.protocolVersion == wakuVersion:
+      debug "Waku peer", peer, wakuVersion
+    else:
+      raise newException(UselessPeerError, "Incompatible Waku version")
+
+Is this what we want? Decide!
+-->
+
+<!-- TODO: Leave room for RECOMMENDATIONS FOR CLIENTS and DEPRECATION NOTICE when relevant -->
+
+
+## Security considerations
+
+There are several security considerations to take into account when running Waku. Chief among them are: scalability, DDoS-resistance and privacy. These also varies depending on what capabilities are used, such as mailserver, light node, and so on.
+
+<!-- TODO: elaborate on security considerations -->
+
+<!-- TODO: Light node security considerations
+
+> Running a node as a light node mode impacts privacy due to the fact that it becomes identifiable what nodes care about if they aren't relaying traffic.
+
+Replace with:
+
+> I think the main privacy concern with light nodes is that the directly connected peers will know that a message originates from them (as it are the only ones it sends). And yes, based on that they can make some assumptions on which messages (topics) they are interested in also.
+
+-->
+
+
+## Implementation Notes
+
+Notes useful for implementing Waku mode.
+
+
+<!-- TODO: Implementation notes, possibly link to matrix and have something similar but lightweight to https://tools.ietf.org/html/rfc8446#appendix-C
+
+Alt, break this out into issue for enhancement -->
+
+<!-- TODO(Dean): Break out into Status spec and remove this section
+
+The golang implementation of Whisper (v.6) already uses packet codes 0x00 - 0x03. Parity's implementation of v.6 will also use codes 0x00 - 0x03. Codes 0x7E and 0x7F are reserved, but still unused and left for custom implementation of Whisper Mail Server.
+
+-->
+
+## Footnotes
+
+1. https://github.com/ethereum/devp2p/blob/master/rlpx.md
+
+
+## Changelog
+
+| Version | Comment |
+| :-----: | ------- |
+| 0.1.0 (current) | Initial Release |
+
+### Differences between shh/6 waku/0
+
+Summary of main differences between this spec and Whisper v6, as described in [EIP-627](https://eips.ethereum.org/EIPS/eip-627):
+
+- RLPx subprotocol is changed from `shh/6` to `waku/0`
+- Light node capability
+- Whisper Mail Server and Whisper Mail Client implemented
+
+<!-- TODO: Document further differences with Whisper v6 -->
+
+## Acknowledgements
+ - kdeme
```

Comment:
@kdeme please ack if you are comfortable with using your real name or not (not public on GH)

Reply:
ack